### ISSUE TITLE
unexpected context behaviour

### ISSUE THREAD

**Issue by bitplorer** [Knio/dominate#173]
Here is a minimal code 
``` python
def subcontext(*args):
    with div("sub context") as _subcontext:
        _subcontext.add(*args)
    return _hello

with div("context") as context_1:
    subcontext(div("inside subcontext"))

with div("context") as context_2:
    inside_context_and_hello = div("inside context and subcontext")
    subcontext(inside_context_and_hello)

print(context_1)
print(context_2)
```
**expected context_1** output :

```html
<div>
  context
  <div>
    sub context
    <div>
      inside subcontext
    </div>
  </div>
</div>
```

**actual context_1** output:
```html
<div>
  context
  <div>
    inside subcontext
  </div>
  <div>
    sub context
    <div>
      inside subcontext
    </div>
  </div>
</div>
```

while **context_2 actual output is same as expected**:
```
<div>
  context
  <div>
    inside context and subcontext
  </div>
  <div>
    sub context
    <div>
      inside context and subcontext
    </div>
  </div>
</div>
```
how should context behaviour be changed ?

**Comment by Knio** [comment]
that does seem weird. what is `_hello` on line 4?

**Comment by Knio** [comment]
case 1 is fixed.
case 2 i do not believe should have different behaviour than case 1. if you want a node to be added twice, it should be explicit with `.add()` in both places

**Comment by Knio** [comment]
published `2.8.0`
